cpp/ql/src/Security/CWE/CWE-191/UnsignedDifferenceExpressionComparedZero.ql

```diff
@@ -33,46 +34,52 @@ predicate isGuarded(SubExpr sub, Expr left, Expr right) {
   )
 }
 
+/**
+ * Gets an expression that is less than or equal to `sub.getLeftOperand()`.
+ * These serve as the base cases for `exprIsSubLeftOrLess`.
+ */
+Expr exprIsLeftOrLessBase(SubExpr sub) {
+  interestingSubExpr(sub, _) and // Manual magic
```

Comment:
Note: There's a subtle change here in the structure of how the `exprIsSubLeftOrLess` predicate is evaluated. On `main` we join with `interestingSubExpr(sub, _)` in every iteration of the recursion. However, by moving this conjunct into `exprIsLeftOrLessBase` we're only joining with `interestingSubExpr(sub, _)` in the base case. This is really what we want since `sub` doesn't change throughout the recursion, so we only need the `interestingSubExpr(sub, _)` conjunct to restrict the base case. Joining with this in any of the recursive cases is wasted work